libs/workspace/src/schematics/dt-component-e2e/index.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DtComponentE2EOptions } from './schema';
+import { strings } from '@angular-devkit/core';
+import { join } from 'path';
+import {
+  Rule,
+  Tree,
+  chain,
+  apply,
+  url,
+  template,
+  mergeWith,
+} from '@angular-devkit/schematics';
+import * as ts from 'typescript';
+import { getSourceFile, findNodes, getIndentation } from '../utils/ast-utils';
+import { InsertChange, commitChanges } from '../utils/change';
+
+interface DtE2EExtendedOptions {
+  selector: string;
+  componentModule: {
+    name: string;
+    package: string;
+  };
+  e2eComponent: {
+    component: string;
+    module: string;
+  };
+  name: string;
+}
+
+function generateComponentOptions(
+  name: string,
+): { name: string; package: string } {
+  return {
+    name: `Dt${strings.classify(name)}Module`,
+    package: `@dynatrace/barista-components/${strings.dasherize(name)}`,
+  };
+}
+
+function generateE2EComponentOptions(
+  name: string,
+): { component: string; module: string } {
+  return {
+    component: `DtE2E${strings.classify(name)}`,
+    module: `DtE2E${strings.classify(name)}Module`,
+  };
+}
+
+/**
+ * Adds a new route inside the ui-test routes
+ */
+export function addRoute(options: DtE2EExtendedOptions): Rule {
+  return (host: Tree) => {
+    const modulePath = join(
+      'apps',
+      'components-e2e',
+      'src',
+      'app',
+      'app.routing.module.ts',
+    );
+    const sourceFile = getSourceFile(host, modulePath);
+
+    /**
+     * find last route and add new route
+     */
+    const routesDeclaration = findNodes(
+      sourceFile,
+      ts.SyntaxKind.VariableDeclaration,
+    ).find(
+      (node: ts.VariableDeclaration) => node.name.getText() === 'routes',
+    ) as ts.VariableDeclaration;
+    const routesElements = (routesDeclaration.initializer as ts.ArrayLiteralExpression)
+      .elements;
+    const lastElement = routesElements[routesElements.length - 1];
+    const end = routesElements.hasTrailingComma
+      ? lastElement.getEnd() + 1
+      : lastElement.getEnd();
+    const indentation = getIndentation(routesElements);
```

Comment:
I would omit the indentation and run a task on as the last step in the schematics that uses prettier to format your changes. a simple `npm run format:write`.

THen you don't have to care about trailing commas indentations and everything.

Reply:
@lukasholzer Please add more information on how to run an npm script as part of the schematic chain :) You have the most knowledge here :)

Reply:
You can take a look how nx is doing it [here](https://github.com/nrwl/nx/blob/master/packages/workspace/src/utils/rules/format-files.ts)

They are shipping this rule so just import it and add it in the end of the chain:

```

import { formatFiles } from '@nrwl/workspace';

..
    return chain([
      mergeWith(templateSource),
      addRoute(extendedOptions),
      formatFiles(),
    ]);
```